app/test/non_determinism_test.go

```diff
@@ -0,0 +1,132 @@
+package app_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/celestiaorg/celestia-app/v2/app/encoding"
+	"github.com/celestiaorg/celestia-app/v2/pkg/appconsts"
+	testutil "github.com/celestiaorg/celestia-app/v2/test/util"
+	"github.com/celestiaorg/celestia-app/v2/test/util/blobfactory"
+	"github.com/celestiaorg/celestia-app/v2/test/util/testfactory"
+	"github.com/celestiaorg/go-square/blob"
+	appns "github.com/celestiaorg/go-square/namespace"
+	"github.com/cosmos/cosmos-sdk/codec"
+	hd "github.com/cosmos/cosmos-sdk/crypto/hd"
+	keyring "github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// TestNonDeterminismBetweenMainAndV1 executes a set of different transactions,
+// produces an app hash and compares it with the app hash produced by v1.x
+func TestNonDeterminismBetweenMainAndV1(t *testing.T) {
```

Comment:
This shouldn't necessary be between v1 and main. It should be comparing the app hash generated from this commit compared to the app hash generated from the previous commit

Reply:
agree that I think we can rename this function since the commitment should be the same across all branches if I'm understanding this comment correctly.

Reply:
[optional] slight tangent: the test could be renamed something like `TestConsistentAppHash` because it is asserting some expected app hash matches some calculated app hash. I think the goal is to identify cases of non-determinism but just because this test passes, it doesn't mean we prevented all potential cases of non-determinism.

Reply:
i agree with all of your point, i only wanted to underline that the hardcoded apphash was generated on v1.x because technically we could do it between any branches but practically the logic for generating apphash lives on v1.x.